container/container_unix.go

```diff
@@ -407,6 +408,29 @@ func (container *Container) TmpfsMounts() []Mount {
 	return mounts
 }
 
+// IntrospectionMounts returns the list of mounts for introspection support
+func (container *Container) IntrospectionMounts() ([]Mount, error) {
+	if !container.HostConfig.Privileged {
+		return nil, nil
+	}
+	path, err := os.Readlink(reexec.Self())
+	if err != nil {
+		return nil, err
+	}
+	return []Mount{
+		{
+			Source:      filepath.Join(filepath.Dir(path), "docker"),
+			Destination: "/usr/bin/docker",
+			Writable:    true,
+		},
+		{
+			Source:      "/run/docker-introspection.sock",
+			Destination: "/run/docker.sock",
```

Comment:
Default socket for docker is `/var/run/docker.sock` so this would not work if image doesn't have a `/var/run -> /run` symlink(like busybox).